tests/common/util/MockLedgerHeaderCache.hpp

```diff
@@ -0,0 +1,35 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2025, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include "data/LedgerHeaderCache.hpp"
+
+#include <gmock/gmock.h>
+#include <xrpl/protocol/LedgerHeader.h>
+
+#include <optional>
+
+struct MockLedgerHeaderCache {
+    MockLedgerHeaderCache() = default;
+    using CacheEntry = data::FetchLedgerCache::CacheEntry;
+
+    MOCK_METHOD(void, put, (CacheEntry), ());
```

Comment:
Now it's inconsistent - in your implemenation above it says `const, here it's not const.
I explained how to make it non-const and don't change lots of code above.